rewrite-hcl/src/main/java/org/openrewrite/hcl/tree/Hcl.java

```diff
@@ -236,6 +237,67 @@ public AttributeAccess withName(HclLeftPadded<Identifier> name) {
         }
     }
 
+    @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+    @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
+    @RequiredArgsConstructor
+    @AllArgsConstructor(access = AccessLevel.PRIVATE)
+    class LegacyIndexAttributeAccess implements Expression, Label {
+        @Nullable
+        @NonFinal
+        transient WeakReference<Padding> padding;
+
+        @With
+        @EqualsAndHashCode.Include
+        @Getter
+        UUID id;
+
+        @With
+        @Getter
+        Space prefix;
+
+        @With
+        @Getter
+        Markers markers;
+
+        @With
+        @Getter
+        Expression base;
+
+        @With
+        @Getter
+        String index;
+
+        @Override
+        public <P> Hcl acceptHcl(HclVisitor<P> v, P p) {
+            return v.visitLegacyIndexAttribute(this, p);
+        }
+
+        @Override
+        public String toString() {
+            return "LegacyIndexAttributeAccess{" + base + "." + index + "}";
+        }
+
+        public Padding getPadding() {
+            Padding p;
+            if (this.padding == null) {
+                p = new Padding(this);
+                this.padding = new WeakReference<>(p);
+            } else {
+                p = this.padding.get();
+                if (p == null || p.t != this) {
+                    p = new Padding(this);
+                    this.padding = new WeakReference<>(p);
+                }
+            }
+            return p;
+        }
+
+        @RequiredArgsConstructor
+        public static class Padding {
+            private final LegacyIndexAttributeAccess t;
```

Comment:
If the type would not have any additional fields we should actually remove it. But we will in fact need it to support that whitespace before the `.` as noted above.